homeassistant/components/zha/select.py

```diff
@@ -107,7 +114,7 @@ def async_restore_last_state(self, last_state) -> None:
         """Restore previous state."""
         if last_state.state and last_state.state != STATE_UNKNOWN:
             self._cluster_handler.data_cache[self._attribute_name] = self._enum[
-                last_state.state.replace(" ", "_")
+                self._translation_keys[last_state.state]
             ]
```

Comment:
This will probably cause issues when coming from a previous HA version? `last_state` might contain a space.
I think we should just keep the `.replace(" ", "_")` for 6 months to ensure every state can be migrated properly(?)

These non-ZCL select entities are a bit special, as they don't rely on zigpy cache (unlike the normal/ZCL ones in `ZCLEnumSelectEntity`).